server/src/main/java/de/materna/dmn/tester/servlets/challenges/ChallengeServlet.java

```diff
@@ -69,13 +113,23 @@ public Response createChallenge(@PathParam("workspace") String workspaceUUID, St
 		Workspace workspace = WorkspaceManager.getInstance().get(workspaceUUID);
 		String uuid = UUID.randomUUID().toString();
 		
-		Challenge bean = (Challenge) SerializationHelper.getInstance().toClass(body, Challenge.class);
-
-		workspace.getChallengeManager().persistFile(uuid, bean);
+		try {
+			Map<String, Object> params = SerializationHelper.getInstance().toClass(body, new TypeReference<HashMap<String, Object>>() {
+			});
+			
+			Challenge challenge = (Challenge) SerializationHelper.getInstance().toClass(body, Challenge.class);
+			
+			challenge = calculateChallengeScenarios(challenge, params);
+			
```

Comment:
I still don't understand this part... Why do we need `params`? Isn't `challenge` already the deserialized version of params?